### PR TITLE
Log levels

### PR DESCRIPTION
**Pull request by kozross**
This adds the ability to specify a log level as well as a tracing mode, both at compile time for scripts, and also when injecting traces into generated code. By doing this, we can log 'different amounts' at the same time (for example, an error code at info level, but a much bigger message at debug level).

Currently, we don't test this new functionality: to ensure CI passes, deprecation warnings have been disabled for the relevant test module, but this needs to be rewritten if this functionality turns out to be beneficial.

### CODE REVIEW

File: Plutarch/Internal/Trace.hs [hunk 1]
```diff
@@ -1,29 +1,73 @@
...
+  forall (a :: S -> Type) (s :: S).
+  Term s PString ->
+  Term s a ->
+  Term s a
+ptraceDebug msg x = pgetConfig $ \case
+  Tracing LogDebug _ -> pforce $ ptrace' # msg # pdelay x
+  _ -> x
+
+{-
 -- | Trace the given message before evaluating the argument.
```
Comment (SeungheonOh) [inline Plutarch/Internal/Trace.hs:69]: We can remove this entirely

**Review by SeungheonOh — COMMENTED** [review]
Looks good to me. I'm wondering, though, currently we only have two log levels--Debug and Info. Will we ever need extra levels of logging in the future?

I think it's no.

File: Plutarch/Internal/Trace.hs (hunk 1 above)
Comment (SeungheonOh) [reply Plutarch/Internal/Trace.hs:69]: nvm, we need to add this back with deprecation warning

**Comment by kozross** [comment]
@SeungheonOh - I don't think so either, but @peter-mlabs is the best judge of this.

**Comment by peter-mlabs** [comment]
Can we also add `pretty` and `eq` instances for `Config` here?

**Comment by kozross** [comment]
`Eq` I can add, but `Pretty` isn't something we use very much in general in Plutarch.

File: Plutarch/Internal/Trace.hs (hunk 1 above)
Comment (kozross) [reply Plutarch/Internal/Trace.hs:69]: What do you mean by this?

Comment (SeungheonOh) [reply Plutarch/Internal/Trace.hs:69]: old `ptrace` function is commented out at the end of the file

**Review by SeungheonOh — APPROVED** [review]
LGTM thanks

**Review by SeungheonOh — APPROVED** [review]

**Pull request merged**